zerver/views/realm.py

```diff
@@ -371,21 +380,39 @@ def update_realm(
         setting_group_id_name = permission_configuration.id_field_name
 
         assert setting_group_id_name in req_group_setting_vars
-
-        if req_group_setting_vars[setting_group_id_name] is not None and req_group_setting_vars[
+        if (
+            req_group_setting_vars[setting_group_id_name] is not None
+            and setting_name in Realm.DIRECT_MESSAGE_PERMISSION_SETTINGS
+        ):
+            setting_value = req_group_setting_vars[setting_group_id_name]
+            new_setting_value = parse_group_setting_value(setting_value["new"])
+
+            expected_current_setting_value = None
+            if setting_value["old"] is not None:
+                expected_current_setting_value = parse_group_setting_value(setting_value["old"])
+
+            current_value = getattr(realm, setting_name)
+            if not validate_group_setting_value_change(
+                current_value, new_setting_value, expected_current_setting_value
+            ):  # nocoverage
+                continue
```

Comment:
We should arrange to have test coverage on this.

Reply:
I have added a test for this.